### ISSUE TITLE
False Positive: "react/prop-types [PROP_NAME] is missing in props validation" when using stateless components + Flow

### ISSUE THREAD

**Issue by sstur** [yannickcr/eslint-plugin-react#467]
The issue is only applicable when I name my function argument "props" but if I change that, the warning goes away. See screenshots:

<img width="705" alt="screenshot 2016-02-25 17 47 11" src="https://cloud.githubusercontent.com/assets/369384/13317540/411e023e-dbe8-11e5-8d9a-6e6f4ef533e6.png">
<img width="713" alt="screenshot 2016-02-25 17 48 07" src="https://cloud.githubusercontent.com/assets/369384/13317545/4522b366-dbe8-11e5-8138-60ad0de35eee.png">

**Comment by sstur** [comment]
Wow, so much fix. So many speed. Thanks @yannickcr!

**Comment by rosskevin** [comment]
I'm seeing this issue with 5.1.1 as not fixed or my configuration is wrong. [Full Repo](https://github.com/rosskevin/my-relay-starter-kit)

Here's my eslintrc:

```json
{
  "parser"  : "babel-eslint",
  "extends" : [
    "standard",
    "standard-react"
  ],
  "env"     : {
    "browser" : true
  },
  "globals" : {
    "__DEV__"      : false,
    "__PROD__"     : false,
    "__DEBUG__"    : false,
    "__COVERAGE__" : false,
    "__BASENAME__" : false
  },
  "rules": {
  }
}
```

```bash
~/p/my-relay-starter-kit ❯❯❯ npm ls eslint-plugin-react                                                                                                                                                                                                                                                                                           ⏎ master ✭ ✱ ◼
my-relay-starter-kit@0.0.1 /Users/kross/projects/my-relay-starter-kit
└── eslint-plugin-react@5.1.1 
```

Do you spot anything I'm doing to misconfigure this?

**Comment by rosskevin** [comment]
FYI - My issue was a bug with flow (nothing with this plugin!): https://github.com/facebook/flow/issues/1908
